src/vs/editor/contrib/stickyScroll/browser/stickyScrollProvider.ts

```diff
@@ -84,14 +92,44 @@ export class StickyLineCandidateProvider extends Disposable {
 		this.onStickyScrollChangeEmitter.fire();
 	}
 
+	private findSumOfRangesOfGroup(outline: OutlineGroup | OutlineElement): number {
+		let res = 0;
+		for (const child of outline.children.values()) {
+			res += this.findSumOfRangesOfGroup(child);
+		}
+		if (outline instanceof OutlineElement) {
+			return res + outline.symbol.range.endLineNumber - outline.symbol.selectionRange.startLineNumber;
+		} else {
+			return res;
+		}
+	}
+
 	private async updateOutlineModel(token: CancellationToken) {
 		if (this._editor.hasModel()) {
 			const model = this._editor.getModel();
 			const modelVersionId = model.getVersionId();
-			const outlineModel = await OutlineModel.create(this._languageFeaturesService.documentSymbolProvider, model, token) as OutlineModel;
+			let outlineModel = await OutlineModel.create(this._languageFeaturesService.documentSymbolProvider, model, token) as OutlineModel;
 			if (token.isCancellationRequested) {
 				return;
 			}
+			// When several possible outline providers
+			if (outlineModel.children.size !== 0 && Iterable.first(outlineModel.children) instanceof OutlineGroup) {
+				if (outlineModel.children.has(this._providerString)) {
+					outlineModel = outlineModel.children.get(this._providerString) as unknown as OutlineModel;
+				} else {
+					let providerString = '';
+					let maxTotalSumOfRanges = 0;
+					for (const [key, outlineGroup] of outlineModel.children.entries()) {
+						const totalSumRanges = this.findSumOfRangesOfGroup(outlineGroup);
+						if (totalSumRanges > maxTotalSumOfRanges) {
+							maxTotalSumOfRanges = totalSumRanges;
+							providerString = key;
+						}
+					}
+					this._providerString = providerString;
+					outlineModel = outlineModel.children.get(this._providerString) as unknown as OutlineModel;
```

Comment:
👮🏼‍♂️ `as unknown as Something` is not allowed and a hint that types are wrong. Either spec the `outlineModel` as or-type or use dedicated variables

Reply:
Yes that commit was just an initial commit. I committed the new changes.